src/ripple/overlay/impl/OverlayImpl.cpp

```diff
@@ -787,9 +879,12 @@ OverlayImpl::crawl()
                     sp->getRemoteAddress().port());
             }
         }
-        auto version = sp->getVersion ();
-        if (! version.empty ())
-            pv[jss::version] = version;
+
+        {
```

Comment:
Why the extra scope? If it's lifetime management, why not just change to:

    if(auto version = sp->getVersion())
        pv[jss::version] = std::move(version);

Reply:
Oh, d'oh.